### PR TITLE
Sma 42 modules specs

### PR DESCRIPTION
**Pull request by livingrockrises**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/modules/src/utils/Utils.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+): string {
+  const userOpHash = keccak256(packUserOp(op, true))
+  const enc = defaultAbiCoder.encode(
+    ['bytes32', 'address', 'uint256'],
+    [userOpHash, entryPointAddress, chainId]
+  )
+  return keccak256(enc)
+}
+
+export function packUserOp(op: UserOperation, forSignature = true): string {
```
Comment (tomarsachin2271) [inline packages/modules/src/utils/Utils.ts:17]: why getUserOpHash and packUserOp methods are in modules utils?? They should be in SA utils.

File: packages/modules/src/interfaces/IValidationModule.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Signer } from 'ethers'
+import { UserOperation } from '@biconomy/core-types'
+import { Bytes } from 'ethers/lib/utils'
+
+export interface IValidationModule {
+  getAddress(): Promise<string>
+  getInitData(): Promise<string>
```
Comment (tomarsachin2271) [inline packages/modules/src/interfaces/IValidationModule.ts:7]: What is the use of getInitData() in IValidationModule? 
It should be in SA which knows about factory address and the validation module address which needs to be used as default validation module.

File: packages/account/src/BiconomySmartAccountV2.ts [hunk 3]
```diff
@@ -210,4 +211,10 @@ export class BiconomySmartAccountV2 extends BaseSmartAccount {
   async getAllSupportedChains(): Promise<SupportedChainsResponse> {
     return this.nodeClient.getAllSupportedChains()
   }
+
+  // async isModuleEnabled(moduleName: string): boolean {
```
[the remark this answers is not in the document]
Comment (livingrockrises) [reply packages/account/src/BiconomySmartAccountV2.ts:215]: yes will update it in V2 implementation PR

File: packages/modules/src/utils/Utils.ts (hunk 1 above)
Comment (livingrockrises) [reply packages/modules/src/utils/Utils.ts:17]: we use it here only. SA package makes read call in the SDK. I can move these to common package maybe

**Review by tomarsachin2271 — APPROVED** [review]

**Pull request merged**